librz/core/cmd/cmd_print.c

```diff
@@ -6570,3 +6428,155 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
 RZ_IPI RzCmdStatus rz_print_value8_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
 	return print_value_size(core, state, argc, argv, 8);
 }
+
+RZ_IPI RzCmdStatus rz_print_pattern0_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	for (int i = 0; i < len; i++) {
+		rz_cons_printf("%02x", 0);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern1_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut8 min = (core->offset & 0xff);
+	for (ut8 i = 0; i < len; i++) {
+		rz_cons_printf("%02x", i + min);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern2_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	ut16 min = (core->offset & 0xffff);
+	for (ut16 i = 0; i < len; i++) {
+		rz_cons_printf("%04x", i + min);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern4_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	ut32 min = (core->offset & UT32_MAX);
+	for (ut32 i = 0; i < len; i++) {
+		rz_cons_printf("%08" PFMT32x, i + min);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern8_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	ut64 min = (core->offset);
+	for (ut64 i = 0; i < len; i++) {
+		rz_cons_printf("%016" PFMT64x, i + min);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
```

Comment:
I suggest to refactor all these function into one that just accept a size argument. They are all essentially the same

Reply:
They aren't the same due to the size and format string used. Of course, they can be generalized but that would introduce unnecessary complexity, IMHO. Lets keep this as is.